### PR TITLE
Add a GitHub workflow to prepare a release pull request

### PR DESCRIPTION
**Pull request by SpecLad**
### Motivation and context

This automates the first half of the current release process. The second half will be implemented by another workflow.

The reason why it can't all be done in a single workflow is that it's useful to let developers inspect what'll go into the release before actually publishing it, and to apply any last-minute fixes, if necessary. It also allows CI to complete for the release PR.

To support the new workflow, add a `--set` option to `update_version.py` that sets the version to a custom value.

### How has this been tested?

Manual testing on my fork of CVAT. See a test run here: <https://github.com/SpecLad/cvat/actions/runs/6459901481>.

### Checklist

- [x] I submit my changes into the `develop` branch
- ~~[ ] I have created a changelog fragment~~ 
- ~~[ ] I have updated the documentation accordingly~~
- ~~[ ] I have added tests to cover my changes~~
- ~~[ ] I have linked related issues (see [GitHub docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))~~
- ~~[ ] I have increased versions of npm packages if it is necessary
  ([cvat-canvas](https://github.com/opencv/cvat/tree/develop/cvat-canvas#versioning),
  [cvat-core](https://github.com/opencv/cvat/tree/develop/cvat-core#versioning),
  [cvat-data](https://github.com/opencv/cvat/tree/develop/cvat-data#versioning) and
  [cvat-ui](https://github.com/opencv/cvat/tree/develop/cvat-ui#versioning))~~

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/opencv/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.

### CODE REVIEW

**Comment by SpecLad** [comment]
CC @azhavoro

File: .github/workflows/prepare-release.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        run:
+          git push -u origin "release-${{ inputs.newVersion }}"
+
+      - name: Create release pull request
+        env:
+          GH_TOKEN: "${{ github.token }}"
+        run: |
+          gh pr create \
+            --base=master \
+            --title="Release v2.7.4" \
```
Comment (kirill-sizov) [inline .github/workflows/prepare-release.yml:51]: I think it supposed to be something like this:
```suggestion
            --title="Release v${{ inputs.newVersion }}" \
```

**Comment by kirill-sizov** [comment]
I also see that it would be nice to have some validation on the version number that prevents increasing one of X, Y or Z on more than 1. Or was it already implemented and I missed it?

File: .github/workflows/prepare-release.yml (hunk 1 above)
Comment (SpecLad) [reply .github/workflows/prepare-release.yml]: Right, thanks for catching it.

**Comment by SpecLad** [comment]
> I also see that it would be nice to have some validation on the version number that prevents increasing one of X, Y or Z on more than 1.

IMO, this isn't needed. There's no requirement that versions _have_ to be sequential, and we might end up wanting to skip a version for whatever reason.

Although now that you mention it, maybe I should at least add a check that the release does not already exist...

**Pull request merged**